cla_public/apps/checker/tests/test_spreadsheet.py

```diff
@@ -0,0 +1,172 @@
+# -*- coding: utf-8 -*-
+import os
+from flask import url_for, session
+import unittest
+import urlparse
+import xlrd
+
+from cla_public import app
+from cla_public.apps.checker.constants import NO, YES
+from cla_public.apps.checker.forms import ProblemForm, AboutYouForm, \
+    YourBenefitsForm, PropertiesForm, SavingsForm, TaxCreditsForm, \
+    IncomeForm, OutgoingsForm
+from cla_public.apps.checker.tests.utils.forms_utils import \
+    ProblemFormMixin, NON_FORM_FIELDS, \
+    CATEGORY_MAPPING, AboutYouFormMixin, BenefitsFormMixin, \
+    PropertiesFormMixin, SavingsFormMixin, TaxCreditsFormMixin, \
+    IncomeFormMixin, OutgoingsFormMixin
+
+
+FILE_PATH = os.path.join(os.path.dirname(__file__), 'data', 'means_test.xlsx')
+
+FORMS = {
+    '/problem': ProblemForm,
+    '/about': AboutYouForm,
+    '/benefits': YourBenefitsForm,
+    '/property': PropertiesForm,
+    '/savings': SavingsForm,
+    '/benefits-tax-credits': TaxCreditsForm,
+    '/income': IncomeForm,
+    '/outgoings': OutgoingsForm,
+}
+
+END_PAGES = {
+    'P': '/result/eligible',
+    'F': '/help-organisations/'
+}
+
+
+class MeansTestEntry(
+    ProblemFormMixin,
+    AboutYouFormMixin,
+    BenefitsFormMixin,
+    PropertiesFormMixin,
+    SavingsFormMixin,
+    TaxCreditsFormMixin,
+    IncomeFormMixin,
+    OutgoingsFormMixin,
+):
+    """Process data row from means test row and return values for each form"""
+    def __init__(self, **values):
+        self.__dict__.update(values)
+
+    def get_value(self, form_class, field):
+        return getattr(self, '%s_%s' % (form_class.__name__.lower(), field))()
+
+    def get_form_data(self, form_class):
+        method_name = '%s_data' % form_class.__name__.lower()
+        if hasattr(self, method_name):
+            return getattr(self, method_name)()
+        return {name: self.get_value(form_class, name)
+                for name in form_class()._fields.iterkeys()
+                if name not in NON_FORM_FIELDS}
+
+    def yes_or_no(self, spreadsheet_value, extra_cond=True):
+        return YES if spreadsheet_value == 'Y' and extra_cond else NO
+
+    def n_to_yes_no(self, n):
+        return YES if self.n_greater_than(n) else NO
+
+    def n_greater_than(self, n, x=0):
+        try:
+            return True if float(n) > x else False
+        except (ValueError):
+            return False
+
+    def number_if_yes(self, n, d):
+        return int(n) if n and int(n) > 0 and d == YES else None
+
+    def get_total_if_partner(self, your_value, partner_value):
+        val = 0.00
+        if your_value:
+            val += float(your_value)
```

Comment:
I don't think there should be any floats used here. Money values should be integers in pence.